src/lib/middleware/auth.js

```diff
@@ -1,43 +1,74 @@
 import util from '../util';
+import conf from '../config';
+import log from '../logger';
+import _ from 'lodash';
+import { APIFetch } from '../api';
 
+const cookie_auth_name = 'feathers-jwt';
+const cookie_user_info = 'user';
 const cookie_name = 'login';
 const cookie_config = {
   maxAge: 30 * 24 * 60 * 60 * 1000,
   httpOnly: true
 };
 
-// Login: ?login=1
-// Logout: ?logout=1
-module.exports = function auth(req, res, next) {
-  const login = req.body.login || req.query.login;
-  const logout = req.body.logout || req.query.logout;
-  // mock login user
-  if (login === '1') {
-    const user_obj = {
-      id: 'user0001',
-      name: 'Test User',
-      team: 'saraban',
-      image: util.site_url('http://lorempixel.com/120/120/cats/')
-    };
-    req.cookies[cookie_name] = user_obj;
-    res.cookie(cookie_name, user_obj, cookie_config);
+function parse_jwt(token) {
+  try {
+    const base64Url = token.split('.')[1];
+    const base64 = base64Url.replace('-', '+').replace('_', '/');
+    return JSON.parse(new Buffer(base64, 'base64').toString());
+  } catch (err) {
+    log.error(err);
+    return null;
   }
+}
 
-  // mock logout user
-  if (logout === '1') {
-    res.clearCookie(cookie_name, cookie_config);
-    delete req.cookies[cookie_name];
+module.exports = function auth(req, res, next) {
+  const jwt = req.cookies[cookie_auth_name];
+  const jwt_data = parse_jwt(jwt);
+  const user_id = jwt_data && jwt_data._id;
+  if (jwt) {
+    const options = {
+      headers: {
+        Authorization: 'Bearer ' + jwt
+      }
+    };
+    req.api = APIFetch(conf.get('service.api.url'), options, _.identity);
   }
 
-  if (req.cookies[cookie_name]) {
-    req.user = res.locals.user = req.cookies[cookie_name];
+  let get_user;
+  if (jwt && user_id) {
+    get_user = req.api('/users/' + user_id);
+  } else {
+    get_user = Promise.resolve(null);
   }
 
-  // require authenticated user
-  if (!req.user) {
-    res.redirect('/login');
-    return;
-  }
+  return get_user
+  .then(data => {
+    if (data) {
+      const user = data;
+      user.token = jwt;
+      user.department = data.departments[0];
```

Comment:
I think `data.departments[0]` should be `data.department`.

REF : https://github.com/youpin-city/youpin-api/commit/e66db4503dcaf2618b70cd551c1976e9607747a5